packages/compass-query-bar/src/components/option-editor.tsx

```diff
@@ -143,6 +143,38 @@ const OptionEditor: React.FunctionComponent<OptionEditorProps> = ({
     });
   }, [schemaFields, serverVersion]);
 
+  const onFocus = (editorView: EditorEventCallbackContext, content: string) => {
+    if (hasAutofix && content === '') {
+      editorView.setContent('{}', 1);
+      onChange('{}');
```

Comment:
You should not be manually calling onChange here, as we dicsussed, similar to autocomplete, the only thing that should happen here is triggering a state change on the editor: the flow basically should look as follows:

update codemirror internal state > EdiorView.updateListener is triggered > onChange callback prop is called > redux state is updated